examples/neural_compressor/text-classification/intent-classification/run_clinc.py

```diff
@@ -0,0 +1,674 @@
+            "with private models)."
+        },
+    )
+
+
+@dataclass
+class OptimizationArguments:
+    """
+    Arguments pertaining to what type of optimization we are going to apply on the model.
+    """
+
+    apply_quantization: bool = field(
+        default=False,
+        metadata={"help": "Whether or not to apply quantization."},
+    )
+    quantization_approach: Optional[str] = field(
+        default=None,
+        metadata={"help": "Quantization approach. Supported approach are static, dynamic and aware_training."},
+    )
+    apply_distillation: bool = field(
+        default=False,
+        metadata={"help": "Whether or not to apply distillation."},
+    )
+    teacher_model_name_or_path: str = field(
+        default=False, metadata={"help": "Path to pretrained model or model identifier from huggingface.co/models"}
+    )
+    quantization_config: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Path to the directory containing the YAML configuration file used to control the quantization and"
+            "tuning behavior."
+        },
+    )
+    distillation_config: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Path to the directory containing the YAML configuration file used to control the distillation"
+            "behavior."
+        },
+    )
+    metric: Optional[str] = field(
+        default=None,
+        metadata={"help": "Metric used for the tuning strategy."},
+    )
+    tolerance_criterion: Optional[float] = field(
+        default=None,
+        metadata={"help": "Performance tolerance when optimizing the model."},
+    )
+    verify_loading: bool = field(
+        default=False,
+        metadata={"help": "Whether or not to verify the loading of the quantized model."},
+    )
+
+
+# Mean Pooling - Take attention mask into account for correct averaging
+def mean_pooling(model_output, attention_mask):
+    token_embeddings = model_output["last_hidden_state"]  # First element of model_output contains all token embeddings
+    input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
+    return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(input_mask_expanded.sum(1), min=1e-9)
+
+
+class SetFitModel(torch.nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.model = model
+        if hasattr(model, "config"):
+            self.config = model.config
+
+    def forward(self, input_ids, attention_mask, token_type_ids=None, *args, **kwargs):
+        if token_type_ids is not None:
+            model_output = self.model(
+                input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+            )
+        else:
+            model_output = self.model(input_ids=input_ids, attention_mask=attention_mask)
```

Comment:
Since teacher model `sentence-transformers/paraphrase-mpnet-base-v2` does not take `token_type_ids` as input argument, while student model `sentence-transformers/paraphrase-MiniLM-L3-v2` does, in order to reuse this model class both for teacher and student model, I added this if else branch.

Reply:
Thanks for the explanation !